respa_admin/views/resources.py

```diff
@@ -467,6 +470,67 @@ def _delete_extra_resource_accessibility(self, resource_accessibility_formset):
 
         ResourceAccessibility.objects.filter(resource=self.object).exclude(pk__in=resource_accessibility_ids).delete()
 
+RESOURCE_RELATED_FIELDS = (
+    'images',
+    'opening_hours',
+    'periods',
+    'accessibility_summaries',
+)
+
+
+@transaction.atomic
+def copy_resource(request, resource_id):
+    source_resource = Resource.objects.prefetch_related(*RESOURCE_RELATED_FIELDS).get(pk=resource_id)
```

Comment:
Not a big deal as the resource should not be normally missing, but using [get_object_or_404()](https://docs.djangoproject.com/en/4.0/topics/http/shortcuts/#get-object-or-404) would be a bit nicer to make sure there will be no uncaught exceptions.

Reply:
Made the suggested changes.